src/map/mob.cpp

```diff
@@ -2700,40 +2699,60 @@ int32 mob_dead(struct mob_data *md, struct block_list *src, int32 type)
 
 	// filter out entries not eligible for exp distribution
 	memset(tmpsd,0,sizeof(tmpsd));
-	for(i = 0, count = 0, mvp_damage = 0; i < DAMAGELOG_SIZE && md->dmglog[i].id; i++) {
-		map_session_data* tsd = nullptr;
-		if (md->dmglog[i].flag == MDLF_SELF) {
+
+	count = 0;
+	mvp_damage = 0;
+	for( i = 0; i < md->dmglog.size(); i++ ){
+		const s_dmglog& entry = md->dmglog[i];
+
+		if( entry.flag == MDLF_SELF ){
 			//Self damage counts as exp tap
 			count++;
 			continue;
 		}
-		tsd = map_charid2sd(md->dmglog[i].id);
+
+		map_session_data* tsd = map_charid2sd( entry.id );
+
 		if (tsd == nullptr)
-			continue; // skip empty entries
+			continue; // skip players that are offline
 		if (tsd->bl.m != m)
 			continue; // skip players not on this map
 		count++; //Only logged into same map chars are counted for the total.
 		if (pc_isdead(tsd))
 			continue; // skip dead players
-		if (md->dmglog[i].flag == MDLF_HOMUN && !hom_is_active(tsd->hd))
-			continue; // skip homunc's share if inactive
-		if (md->dmglog[i].flag == MDLF_PET && (!tsd->status.pet_id || !tsd->pd))
-			continue; // skip pet's share if inactive
 
-		if(md->dmglog[i].dmg > mvp_damage) {
+		switch( entry.flag ){
+			case MDLF_NORMAL:
+				dmgbltypes |= BL_PC;
+				break;
+
```

Comment:
The "break;" already serves pretty much as a clear divider, no need for extra empty line here. We should try to keep the "height" of code as small as possible so you can overview more on a single screen. (But I will rewrite this soon anyway, so you can leave it like this and I'll improve it later.)